### PR TITLE
(MODULES-2799) Build custom types

### PR DESCRIPTION
**Pull request by ferventcoder**
Updating the build to support building custom types
again.

### CODE REVIEW

File: build/dsc.rake [hunk 1]
```diff
@@ -75,9 +82,10 @@ eod
       FileUtils.rm_rf(Dir["#{dsc_resources_path_tmp}/**/.{gitattributes,gitignore,gitmodules}"])
       FileUtils.rm_rf(Dir["#{dsc_resources_path_tmp}/**/*.{pptx,docx,sln,cmd,xml,pssproj,pfx,html,txt,xlsm,csv,png,git,yml,md}"])
 
-      puts "Copying vendored resources from #{dsc_resources_path_tmp}/xDscResources to #{vendor_dsc_resources_path}"
-      FileUtils.cp_r "#{dsc_resources_path_tmp}/xDscResources/.", vendor_dsc_resources_path, :remove_destination => true
-      FileUtils.cp_r "#{dsc_resources_path_tmp}/xDscResources/.", dsc_resources_path
+      puts "Copying vendored resources from #{dsc_resources_path_tmp} to #{vendor_dsc_resources_path}"
+      FileUtils.cp_r "#{dsc_resources_path_tmp}/.", vendor_dsc_resources_path, :remove_destination => true
+      FileUtils.cp_r "#{dsc_resources_path_tmp}/.", dsc_resources_path
+      FileUtils.rm_rf "#{vendor_dsc_resources_path}/PSDesiredStateConfiguration"
```
Comment (ferventcoder) [inline build/dsc.rake]: Choosing to delete the directory after the fact.

**Comment by ferventcoder** [comment]
It doesn't pick up my custom added type for some reason. 

Here's how I went about it. The custom changes and instructions above.

Added cWindowsOS

* From a Windows box - `install-module cWindowsOS` (it's not a type we build, it does have some mof files).
* Copy that off to a temp location that you can reach from non-Windows (for me that is `~/code/temp/dsc_custom_resources`).
* Prep the repo for building
* Run `bundle exec rake dsc:build`
* Run `bundle exec rake dsc:resources:import["~/code/temp/dsc_custom_resources"]`
* Run `bundle exec rake dsc:types:build`

The type shows up in the vendored folder but it doesn't appear to be getting built into custom types yet. This is where I need to pick up and evaluate.

**Comment by ferventcoder** [comment]
@Iristyle @jpogran this is ready.

**Comment by ferventcoder** [comment]
Actually one small change to readme.

**Comment by ferventcoder** [comment]
Okay this is ready.

**Comment by ferventcoder** [comment]
And one more change to the readme.

File: build/dsc.rake (hunk 1 above)
Comment (ferventcoder) [inline build/dsc.rake]: Also ensuring that those are explicitly deleted in case someone attempts to vendor those in.

**Comment by jpogran** [comment]
We have testing this by using it to validate that #151 can deal with non vendored resources, and running it in a VM to see that it builds the path to the manifest file correctly.

**Comment by ferventcoder** [comment]
Travis CI issue looks unrelated but something we should fix.

**Comment by ferventcoder** [comment]
This is holding on #153

**Pull request closed without merging**